cdap-ui/app/cdap/components/AbstractWidget/SqlSelectorWidget/SchemaContainer.tsx

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License'); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+
+import withStyles, { WithStyles } from '@material-ui/core/styles/withStyles';
+import ExpansionPanel from '@material-ui/core/ExpansionPanel';
+import ExpansionPanelSummary from '@material-ui/core/ExpansionPanelSummary';
+import ExpansionPanelDetails from '@material-ui/core/ExpansionPanelDetails';
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
+import Table from '@material-ui/core/Table';
+import TableBody from '@material-ui/core/TableBody';
+import TableCell from '@material-ui/core/TableCell';
+import TableHead from '@material-ui/core/TableHead';
+import TableRow from '@material-ui/core/TableRow';
+import Menu from '@material-ui/core/Menu';
+import MenuItem from '@material-ui/core/MenuItem';
+import Button from '@material-ui/core/Button';
+import Typography from '@material-ui/core/Typography';
+
+import FieldRow from 'components/AbstractWidget/SqlSelectorWidget/FieldRow';
+import If from 'components/If';
+
+import { IParsedInputSchema } from 'components/AbstractWidget/SqlSelectorWidget';
+
+const styles = (theme) => {
+  return {
+    schemaContainer: {
+      margin: '5px 0px',
+    },
+    schemaTableHeader: {
+      fontSize: '13px',
+      borderBottom: '0px',
+      paddingLeft: '0px',
+      paddingRight: '0px',
+    },
+    tableHeaderSelectIcon: {
+      marginLeft: '5px',
+    },
+    tableRowTopBorder: {
+      borderTop: `1px solid ${theme.palette.grey[400]}`,
+    },
+    badgeDanger: {
+      backgroundColor: theme.palette.red[200],
+      color: 'white',
+      fontSize: '9px',
+      marginLeft: '5px',
+    },
+    errorRow: {
+      '& td': {
+        border: '0px',
+      },
+      backgroundColor: '#fde2e2',
+      border: `1px solid ${theme.palette.red[200]}`,
+    },
+  };
+};
+
+interface ISchemaContainerProps extends WithStyles<typeof styles> {
+  stage: IParsedInputSchema;
+  onExpandClick: (arg0: any) => void;
+  onSchemaChange: (arg0: any) => void;
+  aliases: object;
+  errorCount: number;
+}
+
+interface ISchemaContainerState {
+  selectFields: string;
+  menuAnchor: any;
```

Comment:
I think we are losing the benefit of statically typed language if we use `any`. This should be either null or an click event target right? Typescript has a type to represent this DOM element.